### PR TITLE
Task3

### PR DESCRIPTION
**Pull request by noor848**

### CODE REVIEW

File: Conduit/Conduit/Controllers/AuthController.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            _configuration = configuration;
+            _mapper = mapper;
+            _IAuthService = iAuthService;
+            _ipasswordHasher = ipasswordHasher;
+        }
+
+        [HttpPost("SignIn" , Name = "SignInUser")]
+        public async Task<IActionResult> SiginIn([FromBody] SignInModel model)
+        {
+            ///var UserData = _mapper.Map<User>(model);
```
Comment (hayatzsala) [inline Conduit/Conduit/Controllers/AuthController.cs]: remove unnecessary comment

File: Conduit/Conduit/helper/Jwt.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Conduit.helper
```
Comment (hayatzsala) [inline Conduit/Conduit/helper/Jwt.cs]: it's a model not helper, move it to Model directory

File: Conduit/Conduit/Dto/UserD.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿namespace Conduit.Dto
+{
+    public class UserD
+    {
+        
+       /// <summary>
+       /// public string UserId { get; set; }
```
Comment (hayatzsala) [inline Conduit/Conduit/Dto/UserD.cs]: ??

File: Conduit/Conduit/Model/SignInModel.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Conduit.Model
+{
+    public class SignInModel
+    {
+       /// <summary>
+       /// public string Username { get; set; }
```
Comment (hayatzsala) [inline Conduit/Conduit/Model/SignInModel.cs:6]: ??

File: Conduit/Conduit/Controllers/AuthController.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                return BadRequest("Invalid credentials");
+            }
+        }
+
+
+        [HttpPost("SignUp", Name = "CreateUser")]
+        public async Task<IActionResult> SignUp(UserD user)
+        {
+            var UserData = _mapper.Map<User>(user);
+            var UserExist = await _UserRepositry.GetUser(UserData.Email);
```
Comment (hayatzsala) [inline Conduit/Conduit/Controllers/AuthController.cs:75]: Good Validation ❤️

File: Conduit/Conduit.Db/Repositry/UserRepositry.cs [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+                 new User
+                 {
+                     UserId= user.UserId,
+                     Age = user.Age,
+                     Bio = user.Bio,
+                     Email = user.Email,
+                     Password = user.Password,
+                     UserName = user.UserName,
+                     Articles = user.Articles,
```
Comment (hayatzsala) [inline Conduit/Conduit.Db/Repositry/UserRepositry.cs]: no need, check the comments in the other PR

File: Conduit/Conduit.Db/Repositry/UserRepositry.cs [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        {
+            return await _context.Users.Where(s=>s.Email.Equals(Email)).SingleOrDefaultAsync();
+
+        }
+
+
+        public async void updateUserData(User UserTable)
+        {
+            var User = await GetUserById(UserTable.UserId);
+            User.UserName = UserTable.UserName;
```
Comment (hayatzsala) [inline Conduit/Conduit.Db/Repositry/UserRepositry.cs:50]: Mapper can work here btw, and it is more better, even when you have a lot of properties in the obj

File: Conduit/Conduit.Db/Service/IAuthService.cs [hunk 8]
```diff
@@ -0,0 +1,13 @@
+﻿using Conduit.Db;
+using Microsoft.Extensions.Configuration;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+
+namespace Conduit.Service
+{
+    public interface IAuthService
+    {
```
Comment (hayatzsala) [inline Conduit/Conduit.Db/Service/IAuthService.cs]: this interface is not used on the DB project, right? if yes, please remove it

File: Conduit/Conduit.Db/Service/IAuth.cs [hunk 9]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Conduit.Service
+{
+    public interface IAuth
+    {
```
Comment (hayatzsala) [inline Conduit/Conduit.Db/Service/IAuth.cs]: this interface is not used on the DB project, right? if yes, please remove it

**Comment by noor848** [comment]
Conduit #3

**Review by hayatzsala — APPROVED** [review]

**Pull request merged**